### DISCUSSION TITLE
graphql-code-generator vs graphql-cli

### DISCUSSION THREAD

**Discussion by cancan101** [dotansimha/graphql-code-generator#8371] [Q&A] [answered]
What are the differences between this (`graphql-code-generator` and [`graphql-cli`](https://github.com/Urigo/graphql-cli)). Both seem to be maintained by members of The Guild an have a similar set of commands + a plugin system.

and at least in some uses cases there seems to be overlap: https://github.com/Urigo/graphql-cli/issues/1807#issuecomment-901694244

**Comment by charlypoly** [comment, marked as answer]
Hi @cancan101,

The GraphQL CLI `codegen` command is just a shortcut/layer over GraphQL Code Generator.
GraphQL CLI initially aimed to offer a set of features broader than GraphQL Code Generator by providing all the common GraphQL toolbox in one CLI.

When it comes to code generation, the GraphQL Code Generator project is the more up-to-date and go-to solution that we recommend.
